backend/src/main/scala/com/softwaremill/adopttapir/starter/formatting/ProjectFormatter.scala

```diff
@@ -0,0 +1,51 @@
+package com.softwaremill.adopttapir.starter.formatting
+
+import cats.effect.IO
+import com.softwaremill.adopttapir.logging.FLogging
+import com.softwaremill.adopttapir.starter.files.FilesManager
+import com.softwaremill.adopttapir.template.{CommonObjectTemplate, GeneratedFile}
+import org.scalafmt.interfaces.Scalafmt
+
+import java.nio.file.Path
+
+class ProjectFormatter(fm: FilesManager) extends FLogging {
+
+  private val ScalaFileExtensions = Set(".scala", ".sbt")
+
+  private lazy val scalafmt = Scalafmt.create(getClass.getClassLoader)
+
+  def format(gfs: List[GeneratedFile]): IO[List[GeneratedFile]] = {
+    findGeneratedFormatFile(gfs) match {
+      case Some(formatFile) => {
+        IO.blocking(fm.createTempDir())
+          .bracket { tempDirectory =>
+            val formattedFiles = for {
```

Comment:
[minor] no need for the assignment, you can just return the result of the `for` directly